src/main/cpp/src/timezones.cu

```diff
@@ -163,4 +342,50 @@ std::unique_ptr<column> convert_utc_timestamp_to_timezone(column_view const& inp
   return convert_timestamp(input, transitions, tz_index, false, stream, mr);
 }
 
+std::unique_ptr<column> time_add(column_view const& input,
+                                 scalar_i64 const& duration,
+                                 table_view const& transitions,
+                                 size_type tz_index,
+                                 rmm::cuda_stream_view stream,
+                                 rmm::mr::device_memory_resource* mr)
+{
+  auto const type = input.type().id();
+
+  switch (type) {
+    case cudf::type_id::TIMESTAMP_SECONDS:
```

Comment:
Here too. Spark only supports microseconds for timestamps. Why are we supporting seconds and mulliseconds at all?

Reply:
ditto